jgrapht-core/src/main/java/org/jgrapht/alg/scoring/ClosenessCentrality.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * (C) Copyright 2017-2017, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.scoring;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import org.jgrapht.DirectedGraph;
+import org.jgrapht.Graph;
+import org.jgrapht.alg.interfaces.ShortestPathAlgorithm;
+import org.jgrapht.alg.interfaces.ShortestPathAlgorithm.SingleSourcePaths;
+import org.jgrapht.alg.interfaces.VertexScoringAlgorithm;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.alg.shortestpath.FloydWarshallShortestPaths;
+import org.jgrapht.graph.EdgeReversedGraph;
+
+/**
+ * Closeness centrality. Computes the closeness centrality of each vertex of a graph. See
+ * <a href="https://en.wikipedia.org/wiki/Closeness_centrality">wikipedia</a> for the definition of
+ * closeness centrality.
```

Comment:
Can you include the definition of Closeness centrality? It's quite short anyway. There seem to be multiple variants, so better clarify which variant you used such that the user of this algorithm doesn't need to look at the implementation. Something like:
Following the definition of XYZ, the closeness centrality score H(x) of a vertex x is defined as H(x)=\sum_{y \neq x} 1/d(x,y). When normalization is used, H(x)=\sum_{y \neq x} N/d(x,y), where N is the total number of vertices in the graph. For details, see <a href="https://en.wikipedia.org/wiki/Closeness_centrality">wikipedia</a>.